app/code/Magento/Catalog/Block/Product/View/Gallery.php

```diff
@@ -170,10 +171,110 @@ public function getGalleryImagesJson()
         return json_encode($imagesItems);
     }
 
+    /**
+     * Retrieve gallery options in JSON format
+     *
+     * @return string
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     * @SuppressWarnings(PHPMD.NPathComplexity)
+     */
+    public function getGalleryOptionsJson()
+    {
+        $optionItems = null;
+
+        //Special case for gallery/nav which can be the string "thumbs/false/dots"
+        is_bool($this->getVar("gallery/nav")) ?
```

Comment:
gallery/nav is the main configuration setting for the gallery. It defines the following operating modes

thumbs - There is a large main picture, with individual thumbnails next to it where the user can select to view any of the other product images in the main picture area.
false - There is a single main picture. Secondary pictures are not shown.
dots - There is a large main picture. A series of dots at the base of the picture indicate that the user can swipe left or right to move to the next or previous image.

The value of gallery/nav is a string, and can be one of "thumbs" "false" or "dots"

Because of the change to the getVar function made in response to #12285 (or more accurately the config converter functions) if the user enters "thumbs" or "dots" as the value for gallery/nav, then a string value containing "thumbs" or "dots" is returned. However if the user enters "false" as a value for gallery/nav, the getVar function now returns a bool. Prior to #12285 it would have always been a string "false".

Because of this, we must now check. If it is a boolean, we output the boolean value as a string. If it is a string, we just output the string.

Personally, I think we should relook at the resolution to #12285, and i propose after this is merged I will raise a PR with possible revertion and alternative implementation to #12285.